project-templates/ecs/basic/README.md

```diff
@@ -0,0 +1,217 @@
+# Waypoint Project Template Example
+
+This is an example of how to create and use project templating in HCP Waypoint.
+
+Pre-Requisites
+
+AWS Account with Admin Permissions
+
+Terraform Enterprise Account with No Code Module Enabled.
+
+## Foundational Infrastructure Setup
+
+### Terraform Cloud Workspaces
+
+1. (CLI-driven only) Create terraform cloud [workspaces](https://developer.hashicorp.com/terraform/cloud-docs/workspaces/creating) in the TFC User Interface based on the following terraform configurations provided in this example:
+   `terraform-aws-example-microservice-infra`
+   `terraform-aws-example-network`
+   ![img_9.png](../readme-images/create_tfc_workspace_ss.png)
+
+
+2. Go to Settings for the created TFC workspaces, Under `Remote State Sharing` allow `Share with all workspaces in this organization`.
+   ![img_10.png](../readme-images/tfc_remote_sharing_ss.png)
+
+
+3. Create a [global variable set](https://developer.hashicorp.com/terraform/tutorials/cloud-get-started/cloud-workspace-create#create-the-workspace) for your AWS Credentials.
+
+### AWS Steps with Terraform
+
+Create the baseline infrastructure with the following steps in your CLI.
+
+1. Navigate to the `terraform-aws-example-network` directory found in this example and run `terraform init` and the `terraform apply` command.
+
+2. Navigate to the `terraform-aws-example-microservice-infra` directory found in this example and run `terraform init` and the  `terraform apply` command.
+
+
+### Terraform Module Creation
+
+TODO(Teresa): Make this repo public.
+
+1. Fork the `hashicorp/terraform-aws-example-microservice-ecs-all-envs` repository in Github [here](https://github.com/hashicorp/terraform-aws-example-microservice-ecs-allenvs) to your own Github organization.
+
+2. Run the following git commands in the forked repository's working directory.
+
+```shell
+go tag v.0.0.1
+go push --tags
```

Comment:
Did you mean:

And also, why push a patch? Maybe I haven't gotten enough context, but could you explain why you are wanting to tag a new patch?

```suggestion
git push --tags
```

Reply:
Ah! Excellent question. In order for TFC to be able to publish the allenvs as a no-code module, it needs to have a tag. In this case, we are asking the user to fork the repo to their own github account and then they need to push their own tag so TFC can discover it and allow the user to specify which version of their forked repo they would like to use. 

Similarly when the user creates a project template, they will also need to specify the version of the no-code module they would like the project template to use.